CMake/Autodownload.cmake

```diff
@@ -0,0 +1,66 @@
+## This function auto-download mlpack dependencies.
+## You need to pass the LINK to download from, the name of
+## the dependency, and the name of the compressed package such as
+## armadillo.tar.gz
+## At each download, this module set a generic INCLUDE_DIR path,
+## which mean that you need to set the main path for the include
+## directories for each package.
```

Comment:
Armadillo is now `.tar.gz` since it is downloaded from mlpack website, therefore, there is no reason to have `.tar.xz` I will add a comment to mention this.